src/driver/op.rs

```diff
@@ -147,46 +174,92 @@ where
                 me.index = usize::MAX;
                 Poll::Ready(me.data.take().unwrap().complete(cqe))
             }
+            Lifecycle::CompletionList(..) => {
+                unreachable!("No more flag set for SinglCQE")
+            }
         }
     }
 }
 
-impl<T> Drop for Op<T> {
+impl<T, CqeType> Drop for Op<T, CqeType> {
     fn drop(&mut self) {
+        use std::mem;
+
         let mut inner = self.driver.borrow_mut();
-        let lifecycle = match inner.ops.get_mut(self.index) {
-            Some(lifecycle) => lifecycle,
+        let (lifecycle, completions) = match inner.ops.get_mut(self.index) {
+            Some(val) => val,
             None => return,
         };
 
```

Comment:
You could do us all a favor by adding here something to the effect:

// From this point, 1) the lifecycle will be made Ignored because we don't care about further results but more results are still expected, 2) the Op's entry will be removed from the slab and there won't be any further lifecycle, or 3) panic due to an unexpected state being reached.